Tests: Update Package fixtures & tests

Hwesta opened the pull request:
- Pull package-related fixtures into their own package.json fixture file, reorder
- Add tests for `check_fixity`, `download_request` and `extract_file_request`
- Require `extract_file` to have a GET param of relative_path_to_file, handle error

Hwesta committed:
Tests: move Package-specific fixtures to package.json fixture

Hwesta committed:
Tests: add tests for Package.check_fixity

Hwesta committed:
Tests: Split API tests by resource

Hwesta committed:
Fixtures: Group package fixtures by model

Hwesta commented:
Django also supports YAML as a fixture format (as does loaddata/dumpdata that these are generated from).  Do we want to switch to YAML because it's a less picky format and allows comments?

sevein commented:
YAML also adds many complexities around data types, which we probably don't need. Can we just add the supporting documentation in a README file?

Hwesta commented:
I've found YAML to be lightweight and more flexible than JSON, but I don't feel strongly about switching. README seems overkill for the lightweight context-dependent comments I was thinking of, but could be useful, especially as we get more tests/fixtures

Hwesta committed:
API Tests: Add download package tests

Also remove setting up self.client, as Django does that automatically. Added
using fixtures dir from a location.

Hwesta committed:
API Tests: Add download/extract file tests

Hwesta committed:
API: Return error if extract file has no relative_path_to_file

mistydemeo commented:
I could see comments as being a good argument to switch to YAML, as long as we're confident we're unlikely to introduce type issues.

mistydemeo commented on storage_service/locations/tests/test_api.py:
```diff
@@ -61,6 +61,25 @@ def test_cant_move_to_disabled_locations(self):
+
+    def tearDown(self):
+        for entry in os.listdir(self.fixtures_dir):
+            if entry.startswith('tmp'):
```
Would it be possible for us to stub the stuff that actually creates files so we inhibit them being placed on-disk in the first place?

Hwesta replied:
Probably, but I haven't learned how yet. :) How important do you think this is, vs creating/deleting actual files?

mistydemeo replied:
It probably doesn't need to hold this up in the short-term!
